File: src/test/java/sirius/db/text/PatternReplaceProcessorTest.java
/*
 * Made with all the love in the world
 * by scireum in Remshalden, Germany
 *
 * Copyright by scireum GmbH
 * http://www.scireum.de - info@scireum.de
 */

package sirius.db.text;

import org.junit.Test;

import java.util.regex.Pattern;

public class PatternReplaceProcessorTest extends TokenProcessorTest {

    @Test
    public void testTokenizing() {
        assertExactTokenizing("a7b8$c", new PatternReplaceProcessor(Pattern.compile("[^a-z]"), " "), "a b  c");
        assertExactTokenizing("a7b\278$c\10\n\t", PatternReplaceProcessor.createRemoveControlCharacters(), "a7b 8$c   ");
    }
}

File: src/test/kotlin/sirius/db/text/PatternReplaceProcessorTest.kt
/*
 * Made with all the love in the world
 * by scireum in Remshalden, Germany
 *
 * Copyright by scireum GmbH
 * http://www.scireum.de - info@scireum.de
 */

package sirius.db.text

import org.junit.jupiter.api.Test
import java.util.regex.Pattern

/**
 * Tests the [PatternReplaceProcessor].
 */
class PatternReplaceProcessorTest : TokenProcessorTest() {
    @Test
    fun tokenizing() {
        assertExactTokenizing("a7b8\$c", PatternReplaceProcessor(Pattern.compile("[^a-z]"), " "), "a b  c")
        assertExactTokenizing(
            "a7b\u00178\$c\u0008\n\t",
            PatternReplaceProcessor.createRemoveControlCharacters(),
            "a7b 8\$c   "
        )
    }
}
